### DISCUSSION TITLE
Error msg with calcTaylor function

### DISCUSSION THREAD

**Discussion by nargesmbg** [mtex-toolbox/mtex#1982] [Ask Anything]
Hello all, 

I'm trying to calculate the taylor factor for my samples and plot them using following code: 
%% Taylor Factor Calculation for PolyCrystal 
path =  'E:\Automatic Analysis'
cd(path)
mkdir("TaylorFactor")
M_Tay = cell(1, length(files))
%we define first some strain 

strain = 0.2 %define 20% strain
epsilon = strain * strainTensor(diag([0 1 0])) %y axis is our building direction
for i =1: length(files)

    %Ni has the fcc crystal, let's consider fcc slip system
    sS = symmetrise(slipSystem.fcc(grains{i}.CS))
    
    %apply Taylor Model
    [M,b,W] = calcTaylor(inv(grains{i}.meanOrientation)*epsilon,sS); %M is the taylor factor, b is vector of slip rates for all slip system, and W is the spinTensor which means how the orientation is evolved after this strain.
    M_Tay(i) = M 

    figure
    plot(grains{i},M(i))
    mtexColorbar
    mtexColorMap white2black
    name = append(path,'\TaylorFactor\',filenames(i),'_20%strain_010.tif')
    saveFigure(name{1})

end

I'm receiving the following error message: 

Error using optimoptions
Invalid solver specified. Provide a solver name or handle (such as
'fmincon' or @fminunc).
Type DOC OPTIMOPTIONS for a list of solvers.

Error in strainTensor/calcTaylor (line 94)
  options =
  optimoptions('linprog','Algorithm','interior-point-legacy','Display','none');

Error in Automatic (line 121)
    [M,b,W] = calcTaylor(inv(grains{i}.meanOrientation)*epsilon,sS);

could someone help me to figure it out how can I fix it? 
thank you very much 

Best 
Narges

**Comment by nargesmbg** [comment]
hi everyone, can someone help me to fix this problem? 

Best 
Narges

**Comment by kilir** [comment]
Hi,

do you have the Optimization toolbox installed?

Cheers,
Rüdiger
